### PR TITLE
Support cloning multiple git repos

### PR DESCRIPTION
**Pull request by imperfect-fourth**
Closes: #17 
https://buildkite.com/hasura/smooth-checkout-buildkite-plugin-integration-tests/builds/10#2fc893e4-9cfe-4965-a0b8-6fa6a7367afc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,18 +31,38 @@ Allowed values for `ref`:
 - Git tag
 - Commit SHA (40 character long hash)
 
+### Checking out multiple repos
+You can checkout multiple repositories by providing multiple `config` elements:
+```yaml
+steps:
+  - command: echo "Checks out repo from mirror (fall back to github in case of failure)"
```
Comment (scriptnull) [inline README.md:38]: Let us modify the command to echo `Checks out multiple git repositories`

File: hooks/checkout [hunk 2]
```diff
@@ -77,24 +81,44 @@ setup_git_repo() {
...
+    index=$((index+=1))
+  done
+}
+
+PLUGIN_PREFIX="BUILDKITE_PLUGIN_SMOOTH_CHECKOUT"
+
+index=0
+while true; do
+  config_prefix="${PLUGIN_PREFIX}_REPOS_${index}_CONFIG"
+  if [[ -z "$(get_indirected_env "${config_prefix}_0_URL")" || "$index" -gt 5 ]]; then
```
Comment (scriptnull) [inline hooks/checkout:113]: Why this restriction - `"$index" -gt 5`?

**Review by scriptnull — COMMENTED** [review]
Let's move the tests in the playground pipeline to https://buildkite.com/hasura/smooth-checkout-buildkite-plugin-integration-tests

File: hooks/checkout (hunk 2 above)
Comment (imperfect-fourth) [reply hooks/checkout:113]: Oh I forgot to remove this, it was a fallback during testing so that the job doesn't get stuck in an infinite loop

**Comment by imperfect-fourth** [comment]
Updated tests in https://buildkite.com/hasura/smooth-checkout-buildkite-plugin-integration-tests

**Review by scriptnull — APPROVED** [review]

**Pull request merged**